nml/ast/switch.py

```diff
@@ -71,7 +98,8 @@ def get_action_list(self):
 
     def __str__(self):
         var_range = 'SELF' if self.var_range == 0x89 else 'PARENT'
-        return 'switch({}, {}, {}, {}) {{\n{}}}\n'.format(str(next(iter(self.feature_set))), var_range, str(self.name), str(self.expr), str(self.body))
+        params = "" if not self.param_list else "{}, ".format(", ".join(str(x) for x in self.param_list))
+        return 'switch({}, {}, {}, {}{}) {{\n{}}}\n'.format(str(next(iter(self.feature_set))), var_range, str(self.name), params, str(self.expr), str(self.body))
```

Comment:
[you removed the `str()`s but left the `next(iter(self.feature_set))`]

Well, I guess since it's done in several places, it's either fine or needs fixing separately anyway.